xarray/conventions.py

```diff
@@ -438,15 +438,31 @@ def stackable(dim: Hashable) -> bool:
             )
         except Exception as e:
             raise type(e)(f"Failed to decode variable {k!r}: {e}")
-        if decode_coords in [True, "coordinates", "all"]:
+        if decode_coords in [True, "coordinates", "all", "coordinates_strict"]:
```

Comment:
This implies `decode_coords: Literal[True, "coordinates", "all", "coordinates_strict"]`, but the argument says `bool`-only. 
How come mypy hasn't complained about this before?

Reply:
reveal_type(decode_coords) gives: 

`xarray/conventions.py:442: note: Revealed type is "builtins.bool"`

Reply:
yep, which must've been wrong considering possible values in main was `"coordinates"` and `"all"`.

Replace the input type with `decode_coords: Literal[True, "coordinates", "all", "coordinates_strict"]`.